### PR TITLE
Operator Roles

### LINKED ISSUE #28: Consider calculating `role` for each operator
As continued from #27 :

> I think we just need a way to calculate what role each operator is part of, so we can assign the team `role` based on what the players pick.

@redraskal

### PR DESCRIPTION
**Pull request by stnokott**
- use `go:generate` to generate `map[operator]role` structure dynamically, allowing us to update the map without manually changing anything (given the Ubisoft page we are querying - https://www.ubisoft.com/de-de/game/rainbow-six/siege/game-info/operators -  doesn't change its logic. If that happens, we can still manually update the map)
- derive the team roles based on the chosen operators, i.e. if a player chooses a defender operator, we can safely set their team to "Defense"

Note that I kept the `alliance` field discussed in https://github.com/redraskal/r6-dissect/issues/26#issuecomment-1465327214.
I figured you weren't certain yet that it can be removed.

(Messed up a bit in 3153003, so there's a bit of an ugly amend in 221c7f7)

Fixes #28 .

### CODE REVIEW

**Comment by stnokott** [comment]
Maybe run `go:generate` once to see if it works on your end too

**Comment by stnokott** [comment]
Also, might be worth adding `go:generate ./...` to the build job

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	// Hover to view descriptions of existing attributes.
+	// For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+	"version": "0.2.0",
+	"configurations": [
+		{
+			"name": "Launch file",
+			"type": "go",
+			"request": "launch",
+			"mode": "debug",
+			"program": "${workspaceFolder}/main.go",
```
Comment (redraskal) [inline .vscode/launch.json:12]: I'll add .vscode/launch.json to gitignore so personal usage will not be pushed. I like having the generation as a default option

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+			"args": [
+				"Match-2023-03-08_19-39-16-85-R01.rec",
+				"-x",
+				"out.json"
+			]
+		},
```
Comment (redraskal) [inline .vscode/launch.json:18]: I'll add .vscode/launch.json to gitignore so personal usage will not be pushed. I like having the generation as a default option

**Comment by redraskal** [comment]
> Also, might be worth adding `go:generate ./...` to the build job

We really only need to run this when something is manually added/removed. Going to need some tests soon enough with example replay files.

Maybe we could have a test to check if all operators have a role?

**Review by redraskal — APPROVED** [review]

**Comment by stnokott** [comment]
> We really only need to run this when something is manually added/removed. Going to need some tests soon enough with example replay files.

We also need to run it when a new season drops, i.e. when a new operator is added.

> Maybe we could have a test to check if all operators have a role?

I'll add one

**Pull request merged**